posts/2022-04-15-libertydevops-gitjenkinshelmiks.adoc

```diff
@@ -0,0 +1,204 @@
+ibmcloud ks cluster config --cluster $CLUSTER-ID
+kubectl config current-context
+kubectl create -f deploy/deployment.yaml #( simple k8s deployment command )
+kubectl create -f deploy/service.yaml #( simple k8s deployment command )
+kubectl create -f deploy/route.yaml #( simple k8s deployment command )
+----
+
+Make sure that your Kubernetes configuration files are stored in the same git repository as your Jenkinsfile in a sub-directory called `deploy`. Also ensure that the Docker image name in the Kubernetes deployment configuration file updated manually (or programmatically, if it needs to change at run time):
+
+[.imageblock.img_border_light]
+image::/img/blog/deploymentyaml.png[Image reference in deployment yaml,width=40%,align="center"]
+
+When Jenkins has checked out the Java application code for the code build, all the Kubernetes configuration files are also downloaded to the Jenkins workspace so that Jenkins can run the IBM Cloud and Kubernetes commands to connect to the Kubernetes cluster and deploy the application.
+
+See the Kubernetes documentation for other commands:
+link:https://kubernetes.io/docs/reference/kubectl/cheatsheet/[kubectl Cheat Sheet].
+
+== QA testing cloud-native Java applications with Jenkins
+
+Apart from running JUnit test cases along with the code build phase, Jenkins can trigger functional and integration QA test cases automatically after deploying the cloud-native Java application.
+
+Configure the test cases in the Jenkins job and test it manually. Create a remote job identifier authentication token in the "Trigger builds remotely" section under "Build Triggers". Trigger this test case from the Docker "entrypoint" file by using remote rest API call that uses this authentication token as the identifier.
+
+For example, run the following command in a terminal: 
+
+[source]
+----
+curl -I -u <auth-token> https://<jenkins-host>/job/<job-name>/build?token=<remote-job-identifier-authentication-token>
+----
+
+You can generate an authentication token (auth-token) with link:https://www.postman.com[Postman] using the Jenkins login credentials.
+
+== Kubernetes monitoring tools
+
+Kubernetes provides commands to check the application or cluster logs and memory and CPU usage through the commands like: 
+
+[source]
+----
+kubectl logs ..
+cat /sys/fs/cgroup/cpu/cpuacct.usage (after connecting to k8s pod)
+cat /sys/fs/cgroup/memory/memory.usage_in_bytes (after connecting to k8s pod)
+----
+
+There are several applications available that you can integrate with Kubernetes to persist logs and usage statistics, such as link:https://grafana.com/oss/loki/[Grafana] and link:https://prometheus.io[Prometheus]. You can read more in the following articles:
+
+* link:https://medium.com/nerd-for-tech/logging-at-scale-in-kubernetes-using-grafana-loki-3bb2eb0c0872[Grafana with Loki]
+
+* link:https://k21academy.com/docker-kubernetes/prometheus-grafana-monitoring/[Prometheus setup in Kubernetes]
+
```

Comment:
Consider link(s) to Liberty-related assets like https://community.ibm.com/community/user/wasdevops/blogs/don-bourne1/2021/06/26/metrics-and-monitoring-guidance-for-open-liberty-a?

We can consult @donbourne for what is best to reference too.

Thanks.

Reply:
Removed all except "There are several applications available that you can integrate with Kubernetes to persist logs and usage statistics, such as link:https://grafana.com/oss/loki/[Grafana] and link:https://prometheus.io[Prometheus]. "